commit 9ac71d00398674aaec664f30559f0a21d963862f
Author: Cyrill Gorcunov <gorcunov@gmail.com>
Date:   Fri Nov 23 16:43:04 2007 +1100

    [POWERPC] CHRP: Fix possible NULL pointer dereference
    
    This fixes a possible NULL pointer dereference inside of strncmp() if
    of_get_property() fails.
    
    Signed-off-by: Cyrill Gorcunov <gorcunov@gmail.com>
    Signed-off-by: Paul Mackerras <paulus@samba.org>

Backported to Debian's 2.6.18 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.18.orig/arch/powerpc/platforms/chrp/setup.c linux-source-2.6.18/arch/powerpc/platforms/chrp/setup.c
--- linux-source-2.6.18.orig/arch/powerpc/platforms/chrp/setup.c	2006-09-19 21:42:06.000000000 -0600
+++ linux-source-2.6.18/arch/powerpc/platforms/chrp/setup.c	2008-02-13 21:01:52.000000000 -0700
@@ -105,7 +105,7 @@ void chrp_show_cpuinfo(struct seq_file *
 	seq_printf(m, "machine\t\t: CHRP %s\n", model);
 
 	/* longtrail (goldengate) stuff */
-	if (!strncmp(model, "IBM,LongTrail", 13)) {
+	if (model && !strncmp(model, "IBM,LongTrail", 13)) {
 		/* VLSI VAS96011/12 `Golden Gate 2' */
 		/* Memory banks */
 		sdramen = (in_le32(gg2_pci_config_base + GG2_PCI_DRAM_CTRL)
@@ -192,14 +192,20 @@ static void __init sio_fixup_irq(const c
 static void __init sio_init(void)
 {
 	struct device_node *root;
+	const char *model;
 
-	if ((root = find_path_device("/")) &&
-	    !strncmp(get_property(root, "model", NULL), "IBM,LongTrail", 13)) {
+	root = find_path_device("/");
+	if (!root)
+		return;
+
+	model = get_property(root, "model", NULL);
+	if (model && !strncmp(model, "IBM,LongTrail", 13)) {
 		/* logical device 0 (KBC/Keyboard) */
 		sio_fixup_irq("keyboard", 0, 1, 2);
 		/* select logical device 1 (KBC/Mouse) */
 		sio_fixup_irq("mouse", 1, 12, 2);
 	}
+
 }
 
 
